source/DmapSupport.cpp

```diff
@@ -372,19 +395,79 @@ HRESULT SendIOControlCodeToController(HANDLE & handle, IOControlCode^ iOControlC
     }
 
     auto workItem = ref new WorkItemHandler(
-        [device, iOControlCode, ioControlCompleted, &hr]
+        [device, iOControlCode, ioControlCompleted, &hr, &bufferToDriver, &bufferFromDriver]
     (IAsyncAction^ workItem)
     {
-        create_task(device->SendIOControlAsync(iOControlCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
+        create_task(device->SendIOControlAsync(iOControlCode, bufferToDriver, bufferFromDriver)).then([&hr, &ioControlCompleted](task<unsigned int> t)
```

Comment:
The handle isn't being changed, what it represents is, so this should work.  It would probably be clearer to have it consistent.